for_workspace/cmake_build.bzl

```diff
@@ -38,10 +38,11 @@ cmake_tool = rule(
     attrs = {
         "cmake_srcs": attr.label(mandatory = True),
     },
-    fragments = ["cpp"],
+    host_fragments = ["cpp"],
```

Comment:
What does `fragments` -> `host_fragments` do?

Reply:
This will access the `cpp` fragments in the `host` configuration rather than `target` since this rule will always be building make for the `host` (or `exec`) configuration. Which actually means that I've got the `"_cc_toolchain"` attribute wrong now I think about it - it should be `current_cc_host_toolchain` rather than `current_cc_toolchain`?

I wasn't 100% sure on this though; we may want to rely on the fact that this will referenced via a host configuration and so the toolchain should be set correctly anyway and we shouldn't need to explicitly select the host toolchain. 

Either way these rules aren't actually properly using the toolchain so this area does need some improvement.